storage_controller/src/service.rs

```diff
@@ -2918,6 +2922,68 @@ impl Service {
         .await?
     }
 
+    pub(crate) async fn tenant_timeline_archival_config(
+        &self,
+        tenant_id: TenantId,
+        timeline_id: TimelineId,
+        req: TimelineArchivalConfigRequest,
+    ) -> Result<(), ApiError> {
+        tracing::info!(
+            "Setting archival config of timeline {tenant_id}/{timeline_id} to '{:?}'",
+            req.state
+        );
+
+        let _tenant_lock = trace_shared_lock(
+            &self.tenant_op_locks,
+            tenant_id,
+            TenantOperations::TimelineArchivalConfig,
+        )
+        .await;
+
+        self.tenant_remote_mutation(tenant_id, move |targets| async move {
+            if targets.is_empty() {
+                return Err(ApiError::NotFound(
+                    anyhow::anyhow!("Tenant not found").into(),
+                ));
+            }
+
+            async fn config_one(
+                tenant_shard_id: TenantShardId,
+                timeline_id: TimelineId,
+                node: Node,
+                jwt: Option<String>,
+                req: TimelineArchivalConfigRequest,
+            ) -> Result<(), ApiError> {
+                tracing::info!(
+                "Setting archival config of timeline on shard {tenant_shard_id}/{timeline_id}, attached to node {node}",
+            );
+
+                let client = PageserverClient::new(node.get_id(), node.base_url(), jwt.as_deref());
+                client
+                    .timeline_archival_config(tenant_shard_id, timeline_id, &req)
+                    .await
+                    .map_err(|e| passthrough_api_error(&node, e))
+            }
+
+            // no shard needs to go first/last; the operation should be idempotent
+            // TODO: it would be great to ensure that all shards return the same error
```

Comment:
```suggestion
```

this means much less for timeline archival than timeline ancestor detach.

Reply:
as said below, I think this should remain as TODO

Reply:
Yeah I still kinda feel icky about this because there is also a helper to do the HTTP request to all shards which would give this ability.... but yeah.

Because we don't have any cancelled requests warnings in `allowed_error` I guess this sending a request only to one pageserver without waiting for others is never triggered then in the test as is?

Reply:
yeah we do requests one-by-one and return on the first error, assuming the others would error too. We'd probably have to do requests in parallel or something, but I don't feel this is in the scope of this PR.